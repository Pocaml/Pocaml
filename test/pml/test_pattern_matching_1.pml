let test_pattern_matching lst =
  match lst with
    | x :: xs ->
      match x with
	  | 1	  -> "int    list   (1 :: ints)"
	  | true  -> "bool   list   (true  :: bools)"
	  | false -> "bool   list   (false :: bools)"
	  | 'a'	  -> "char   list   ('a' :: chars)"
	  | ()	  -> "unit   list"
	  | "pml" -> "string list   (\"pml\" :: strings)"
	  | _	  -> "other  list"
    | [] -> "empty list ([])"

let lsts = [
  [1; 2];
  [true; false];
  [false; true];
  ['a'; 'b'];
  [()];
  ["pml"; "cool"];
  [2;1]
]

let _ = list_map (fun lst -> print_endline (test_pattern_matching lst)) lsts
