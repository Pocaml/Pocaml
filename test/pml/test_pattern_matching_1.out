int    list   (1 :: ints)
int    list   (1 :: ints)
bool   list   (false :: bools)
char   list   ('a' :: chars)
unit   list
string list   ("pml" :: strings)
other  list
